crates/oxc_linter/src/lib.rs

```diff
@@ -9,22 +9,47 @@ mod context;
 mod disable_directives;
 mod fixer;
 mod globals;
+mod lint_adapter;
 pub mod rule;
 mod rules;
 
-use std::{fs, io::Write, rc::Rc};
+use std::{collections::BTreeMap, fs, io::Write, rc::Rc, sync::Arc};
 
-pub use fixer::{FixResult, Fixer, Message};
+pub use fixer::{Fixer, Message};
+use lazy_static::lazy_static;
+use lint_adapter::{InputQuery, LintAdapter};
+use miette::{miette, LabeledSpan};
 pub(crate) use oxc_semantic::AstNode;
 use oxc_semantic::Semantic;
 use rustc_hash::FxHashMap;
+use trustfall::{execute_query, Schema, TransparentValue};
 
 use crate::context::LintContext;
 pub use crate::{
     rule::RuleCategory,
     rules::{RuleEnum, RULES},
 };
 
+lazy_static! {
```

Comment:
Not really actionable in this PR, but as a future work item, consider switching to `OnceCell` either as a [built-in if using MSRV Rust 1.70](https://doc.rust-lang.org/stable/std/cell/struct.OnceCell.html) or as [a separate crate](https://docs.rs/once_cell/latest/once_cell/).